crates/noirc_evaluator/src/ssa/ssa_gen/mod.rs

```diff
@@ -118,8 +120,20 @@ impl<'a> FunctionContext<'a> {
         match literal {
             ast::Literal::Array(array) => {
                 let elements = vecmap(&array.contents, |element| self.codegen_expression(element));
-                let typ = Self::convert_non_tuple_type(&array.typ);
-                self.codegen_array(elements, typ)
+                let contents_typ = Self::convert_non_tuple_type(&array.typ);
+                
+                let typ = Self::convert_type(&array.typ);
+                if typ.count_leaves() > 1 {
+                    let slice_length = ast::Literal::Integer(
+                        (array.contents.len() as u128).into(),
+                        ast::Type::Field,
+                    );
+                    let slice_length = self.codegen_literal(&slice_length);
```

Comment:
```suggestion
                    let slice_length = self.builder.field_constant(array.contents.len() as u128);
```
May need to put the length on a different line if the formatter has issues with this.

Reply:
did this suggestion while enacting other changes to this match case